### DISCUSSION TITLE
Clear documentation on https SSL

### DISCUSSION THREAD

**Discussion by jshapiro2** [postalserver/postal#2516] [Help with using Postal]
Hello, 

I have read everything possible.  There is no clear documentation on setting up a third party ssl cert so the admin dashboard will work correctly with https.  Can someone please point me in the correct direction?  I am using Caddy.

**Comment by sinnbeck** [comment]
I can share my Caddyfile so you can do the same (the file is in /opt/postal/config/Caddyfile). After setting it up, just run the command shown here: https://docs.postalserver.io/install/installation/#caddy

I have changed the key type so I can use it for the smtp server as well
```
{
  key_type rsa4096
}

# Update this to match the hostname you wish to use for accessing the
# Postal web interface.
postal.mydomain.com {
  reverse_proxy 127.0.0.1:5000
}

**Reply by jshapiro2** [reply]
thanks

**Comment by willpower232** [comment]
As mentioned above, Caddy is based around generating its own certificates, I'm not sure you can ask Caddy to use an externally created certificate so if you must then you could swap Caddy out for nginx.

Caddys docs are here https://caddyserver.com/docs/
nginx docs are here https://nginx.org/en/docs/

**Reply by jshapiro2** [reply]
I did finally get it to work.  The key was to point caddy to the volume where the keys are auto renewed.
docker run -d \
   --name postal-caddy \
   --restart always \
   --network host \
   -v /opt/postal/config/Caddyfile:/etc/caddy/Caddyfile \
   -v /opt/postal/caddy-data:/data \
   -v /etc/letsencrypt/live/[DOMAIN]/fullchain.pem:/etc/caddy/ssl/fullchain.pem \
   -v /etc/letsencrypt/live/[DOMAIN]/privkey.pem:/etc/caddy/ssl/privkey.pem \
   caddy

**Then just updated the Caddyfile**
# Update this to match the hostname you wish to use for accessing the
# Postal web interface.
[DOMAIN] {
    reverse_proxy 0.0.0.0:5000
    tls /etc/caddy/ssl/fullchain.pem /etc/caddy/ssl/privkey.pem
}

click.[DOMAIN] {
    reverse_proxy 0.0.0.0:5000 {
        header_up X-Postal-Track-Host "1"
    }
    tls /etc/caddy/ssl/fullchain.pem /etc/caddy/ssl/privkey.pem
}

click.[DOMAIN2] {
    reverse_proxy 0.0.0.0:5000 {
        header_up X-Postal-Track-Host "1"
    }
    tls /etc/caddy/ssl/fullchain.pem /etc/caddy/ssl/privkey.pem
}
